File: Leetcode/leetcode-1695-MaximumErasureValue-[array]-[medium]/Solution.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        HashSet<Integer> visited = new HashSet<>();

        int[] prefix = new int[nums.length];
        prefix[0] = nums[0];
        for (int i = 1; i < nums.length; i++) {
            prefix[i] = prefix[i - 1] + nums[i];
        }

        int sum = 0;
        int i = 0, j = 0;
        while (j < nums.length) {
            while (i < nums.length && j < nums.length && !visited.contains(nums[j])) {
                sum = Math.max(sum, prefix[j] - prefix[i] + nums[i]);
                visited.add(nums[j]);
                j++;
            }

            while (i < nums.length && j < nums.length && visited.contains(nums[j])) {
                visited.remove(nums[i]);
                i++;
            }
        }

        return sum;
    }
}

File: Leetcode/leetcode-1695-MaximumErasureValue-[array]-[medium]/Solution.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        prefix = [0] * len(nums)
        prefix[0] = nums[0]
        for i, num in enumerate(nums):
            if (i >= 1):
                prefix[i] = prefix[i - 1] + num
        
        visited = set()

        sum = 0
        i = 0
        j = 0
        while (j < len(nums)):
            while (j < len(nums) and i < len(nums) and nums[j] not in visited):
                sum = max(sum, prefix[j] - prefix[i] + nums[i])
                visited.add(nums[j])
                j = j + 1

            while (i < len(nums) and j < len(nums) and nums[j] in visited):
                visited.remove(nums[i])
                i = i + 1

        return sum